Description: Make folder-color work under python3
 The nautilus-python package dropped support for python2, so for this extension
 to continue to work in groovy and later, we need to make sure it runs with
 python3.
Author: Łukasz 'sil2100' Zemczak <lukasz.zemczak@canonical.com>
Forwarded: no
Last-Update: 2020-06-02
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/nautilus-extension/folder-color.py
+++ b/nautilus-extension/folder-color.py
@@ -293,7 +293,7 @@
                 continue
             
             # Get metadata file/folder
-            item_path = urllib.unquote(item.get_uri()[7:])
+            item_path = urllib.parse.unquote(item.get_uri()[7:])
             item = Gio.File.new_for_path(item_path)
             info = item.query_info('metadata', 0, None)
             # If any metadata > restore menu
@@ -312,7 +312,7 @@
             if each_item.is_gone():
                 continue
             
-            item_path = urllib.unquote(each_item.get_uri()[7:])
+            item_path = urllib.parse.unquote(each_item.get_uri()[7:])
             self.foldercolor.set_color(item_path, self._get_skel_folder(item_path, icon)['name'])
     
     def _menu_activate_restore_all(self, menu, items):
@@ -321,7 +321,7 @@
             if each_item.is_gone():
                 continue
             
-            item_path = urllib.unquote(each_item.get_uri()[7:])
+            item_path = urllib.parse.unquote(each_item.get_uri()[7:])
             self.foldercolor.restore_color(item_path)
             self.foldercolor.restore_emblem(item_path)
     
@@ -331,7 +331,7 @@
             if each_item.is_gone():
                 continue
             
-            item_path = urllib.unquote(each_item.get_uri()[7:])
+            item_path = urllib.parse.unquote(each_item.get_uri()[7:])
             self.foldercolor.restore_emblem(item_path)
     
     def _menu_activate_emblem(self, menu, emblem, items):
@@ -340,7 +340,7 @@
             if each_item.is_gone():
                 continue
             
-            item_path = urllib.unquote(each_item.get_uri()[7:])
+            item_path = urllib.parse.unquote(each_item.get_uri()[7:])
             self.foldercolor.set_emblem(item_path, emblem)
     
     def _menu_activate_custom_color(self, menu, icon, items):
@@ -353,7 +353,7 @@
             if each_item.is_gone():
                 continue
             
-            item_path = urllib.unquote(each_item.get_uri()[7:])
+            item_path = urllib.parse.unquote(each_item.get_uri()[7:])
             custom_icon = self._get_skel_folder(item_path, icon)
             
             src_file = custom_icon['filename']
